src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/sms/AppellantRemoveRepresentationPersonalisationSms.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.appellant.sms;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition.*;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.SmsNotificationPersonalisation;
+import uk.gov.hmcts.reform.iacasenotificationsapi.infrastructure.CustomerServicesProvider;
+
+
+@Service
+public class AppellantRemoveRepresentationPersonalisationSms implements SmsNotificationPersonalisation {
+
+    private final String removeRepresentationAppellantSmsTemplateId;
+    private final CustomerServicesProvider customerServicesProvider;
+
+
+    public AppellantRemoveRepresentationPersonalisationSms(
+        @Value("${govnotify.template.removeRepresentation.appellant.sms}") String removeRepresentationAppellantSmsTemplateId,
+        CustomerServicesProvider customerServicesProvider
+    ) {
+        this.removeRepresentationAppellantSmsTemplateId = removeRepresentationAppellantSmsTemplateId;
+        this.customerServicesProvider = customerServicesProvider;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return removeRepresentationAppellantSmsTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(AsylumCase asylumCase) {
+
+        return Collections.singleton(asylumCase
+            .read(MOBILE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("appellantMobileNumber is not present")));
+    }
+
+    @Override
+    public String getReferenceId(Long caseId) {
+        return caseId + "_REMOVE_REPRESENTATION_APPELLANT_SMS";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(AsylumCase asylumCase) {
```

Comment:
you can use different method with callback object
```
Map<String, String> getPersonalisation(Callback<AsylumCase> callback)
```
so you don't need to add additional field `CCD_CASE_ID`

Reply:
Now using the above method and I've also moved the unnecessary CCD_CASE_ID field.